packages/api/amplify_api/example/integration_test/graphql/api_key_test.dart

```diff
@@ -105,6 +105,66 @@ void main({bool useExistingTestUser = false}) {
             expect(blogFromEvent?.name, equals(name));
           });
 
+          testWidgets(
+              'subscribe() should handle multiple streams synchronously',
+              (WidgetTester tester) async {
+            final readyCompleter = Completer<void>();
+            final readyCompleter2 = Completer<void>();
+            final dataCompleter = Completer<Blog>();
+            final dataCompleter2 = Completer<Blog>();
+
+            final name =
+                'Integration Test Blog - subscription sync test ${uuid()}';
+
+            final subscriptionRequest1 = ModelSubscriptions.onCreate(
+              Blog.classType,
+              authorizationMode: APIAuthorizationType.apiKey,
+            );
+
+            final stream1 = Amplify.API.subscribe(
+              subscriptionRequest1,
+              onEstablished: expectAsync0(readyCompleter.complete),
+            );
+            stream1.listen(
+              ((event) {
+                if (event.data?.name == name) {
+                  final blogFromEvent = event.data;
+                  expect(blogFromEvent?.name, equals(name));
+                  dataCompleter.complete(event.data);
+                }
+              }),
+              onError: (Object e) => fail('Error in subscription stream: $e'),
+            );
+
+            final subscriptionRequest2 = ModelSubscriptions.onCreate(
```

Comment:
nit: Could the 2nd request be `onUpdate` so the use case makes a little more sense? As written, both these streams have the same output so kinda strange use case to test (though it does effectively test the bug this PR is fixing).

Reply:
Swapped to `onDelete` for continuity.